src/usecases/retrieveAverageParticipantsInVisit.js

```diff
@@ -0,0 +1,23 @@
+const retrieveAverageParticipantsInVisit = ({ getDb }) => async (trustId) => {
+  if (!trustId) return { error: "A trustId must be provided." };
+
+  const db = await getDb();
+  const [{ average_participants: averageParticipantsInVisit }] = await db.any(
+    `SELECT AVG(participants) as average_participants
```

Comment:
If there are 0 events in the database, I believe this query will return null as participants will be NULL and AVG removes all NULL results before calculating the average so it will also return NULL.

One option could be to use COALESCE to default the result of the AVG query to 0 if it returns NULL 
e.g. `SELECT COALESCE(AVG(participants), 0) as average_participants`

Reply:
Nice spot, updated and added a test for this as well.